### ISSUE TITLE
observer notification only if transaction got commited

### ISSUE THREAD

**Issue by slawomirpiotrowski** [dotnet/orleans#8767]
Is this possible to use Observer in a way that would send notification when current transaction gets commited (avoid notification if transaction gets rolled back)?

**Comment by temcewen** [comment]
I don't think so unfortunately. I would personally love to be able to hook into the lifecycle of a transaction, but I don't think you can.

**Comment by slawomirpiotrowski** [comment]
I was able to work aroud that limitation by sending a message (using masstransit) to myself from within transaction. When receiving that message I check (using grain method using performread) if transaction was commited. Using performread causes a wait if transaction is still not commited (nor rolled back). Then, inside performread I read grain state to check if transaction was actually commited and proceed only if it was.

An ugly workaround given that Orleans already knows if transaction was commited or rolled back, but oh well - at least it works.

In some usages it might be problematic, that few writes might have happened before masstransit message was delivered. In my application it's not a problem.
